### ISSUE TITLE
[Bug]: React-Native iOS Mobile Application Disconnects after some seconds

### ISSUE THREAD

**Issue by NoisyBoyII** [MetaMask/metamask-sdk#689]
### SDK

React-Native

### Provide environment information

I am using react-native for creating Mobile Application and testing it on iOS 17.2.1.My react version is "18.2.0" &
react-native version is "0.72.6" and using metamask mobile application for user authentication and other functionalities.

### MetaMask SDK Version

i am using "@metamask/sdk-react": "^0.15.0"

### MetaMask Mobile app Version

v7.15.0

### What browser are you using? (if relevant)

Not Relevant

### How are you deploying your application? (if relevant)

Testing Locally

### Describe the Bug

I am calling get Provider and creating a smart contract variable with abi , address & signer and doing call with metamask but when i am storing this provider into useState it's getting expired or null after like 30 seconds or 40 seconds  because i am not able to see account when i do console.log and getting error message when calling smart contract method(Error Message below the code).
After manually opening Metamask application .. everything starts working again but after few seconds i am seeing the below message.
```
   const provider = sdk.getProvider();
    console.log(provider);
    const web3Provider = new ethers.providers.Web3Provider(provider);
    const signer = web3Provider?.getSigner();
    newProductContract = new ethers.Contract(
      productContractAddress,
      productContractABI,
      signer
    );
    return newProductContract;
```
    
    [Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x7EB6Ea91BCA876aAdC7eC2B8b862FBAbc3C1038B","to":"0xf6376D9699c16cDE0a1ee0a333E24Ac425017283","data":"0x4e2327780000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={}, code=CALL_EXCEPTION, version=providers/5.7.2)]

creating metamask sdk like this
```
const sdk = new MetaMaskSDK({
  openDeeplink: (link: string) => {
    if (canOpenLink) {
      Linking.openURL(link);
    }
  },
  timer: BackgroundTimer, // To keep the dapp alive once it goes to background.
  dappMetadata: {
    name: "dAPP", // The name of your dapp.
    url: "http://example.com", // The URL of your website.
  },
});
```

Questions

1. Do anyone have experience this bug?
2. Does metamask disconnect after few seconds ? if yes please can anyone tell me how to identify if its disconnect in react-native so that i can ask user to reconnect.
3. May this happening due to breaking of socket connection? because when i open metamask application again from background i got this message
SocketService::on 'clients_connected' skip sending originatorInfo on pause

### Expected Behavior

it should be connected for 1 or 2 mintues.

### To Reproduce

I have provided the code above please check and reproduce. if any assistance needed please message me.

**Comment by abretonc7s** [comment]
Hi @NoisyBoyII ,

could you please provide a minimal repo to reproduce the issue?
It seems that you are generating a call to blockchain while the wallet is sleeping which can prevent the error.
A workaround is to add an infura api key to your code https://docs.metamask.io/wallet/how-to/make-read-only-requests/ .

Please, let us know if that helps

**Comment by NoisyBoyII** [comment]
Hi @abretonc7s ,
I have added Infura API key to my code and its working great for read calls.
I wanted to know like after some seconds without interaction with metamask .. socket gets close by default so is there workaround to know that metamask connection is disconnected?
Right now i am using this but no response .. is there any method we can know if its disconnected ?
```
        ethereum.on("disconnect", () => {
          console.log('useEffect::ethereum on "disconnect"');
          setConnected(false);
        });
```

**Comment by christopherferreira9** [comment]
Closing this issue for the time being as it is providing the expected behaviour and the new `paused` feature will be available on version `0.16.0`.
